src/tests/modify.test.js

```diff
@@ -298,3 +306,87 @@ describe('modify() - basic mutations', () => {
     });
   });
 });
+
+describe('modify() - reoder fields', () => {
+  it('reorder fields - basic usage', () => {
+    const baseExample = {
+      properties: {
+        /* does not matter */
+      },
+      'x-jsf-order': ['field_a', 'field_b', 'field_c', 'field_d'],
+    };
+    const result = modify(baseExample, {
+      order: {
+        fields: ['field_c', 'field_a', 'field_b'],
+      },
+    });
+
+    // 💡 Note how the missing field (field_d) was added to the end as safety measure.
+    expect(result).toMatchObject({
+      'x-jsf-order': ['field_c', 'field_a', 'field_b', 'field_d'],
+    });
+  });
+
+  it('reorder fields -  based on originalOrder', () => {
+    const baseExample = {
+      properties: {
+        /* does not matter */
+      },
+      'x-jsf-order': ['field_a', 'field_b', 'field_c', 'field_d'],
+    };
+    const result = modify(baseExample, {
+      order: (original) => {
+        return {
+          fields: original.reverse(),
+        };
+      },
+    });
+
+    expect(result).toMatchObject({
+      'x-jsf-order': ['field_d', 'field_c', 'field_b', 'field_a'],
+    });
+  });
+
+  it('reorder fields in fieldsets', () => {
+    const baseExample = {
+      properties: {
+        address: {
+          properties: {
+            /* does not matter */
+          },
+          'x-jsf-order': ['first_line', 'zipcode', 'city'],
+        },
+        age: {
+          /* ... */
+        },
+      },
+      'x-jsf-order': ['address', 'age'],
+    };
+
+    const result = modify(baseExample, {
+      fields: {
+        address: {
+          order: (original) => {
+            return {
+              fields: original.reverse(), // ['city', 'zipcode', 'first_line']
+            };
+          },
```

Comment:
It would be nice to include a test with the non-function version of order for fields' order

```
address: {
  order: {
    fields: ['city', 'zipcode', 'first_line']
  }
}
```